src/core/reactor_backend.cc

```diff
@@ -483,7 +483,12 @@ bool reactor_backend_aio::await_events(int timeout, const sigset_t* active_sigma
     bool did_work = false;
     int r;
     do {
+        const bool may_sleep = !tsp || (tsp->tv_nsec + tsp->tv_sec > 0);
+        const auto before_getevents = may_sleep ? sched_clock::now() : sched_clock::time_point{};
         r = io_pgetevents(_polling_io.io_context, 1, batch_size, batch, tsp, active_sigmask);
+        if (may_sleep) {
+            _r._total_sleep += sched_clock::now() - before_getevents;
+        }
```

Comment:
How come only one backend gets this adjustment?

Reply:
Probably because it's the one we use currently. This change is from a year or more ago and was dusted off. Let me check the other backends.

Reply:
Right, it's slowly coming back: updating the other reactor backends was part of why I never upstreamed this in the first place. Doing that now.

Reply:
In the last push this is fixed for all 3 reactor backends.